test/interceptors/request.test.ts

```diff
@@ -176,6 +176,21 @@ describe('test request interceptor', () => {
     expect(response.data).toBe(true);
   });
 
+  it('tests must revalidate handling with etag custom e-tag', async () => {
+    const axios = mockAxios({}, { etag: 'fakeEtag', 'cache-control': 'must-revalidate' });
```

Comment:
it would be best to not send etag back as header, to ensure it works with custom etag on its own. (however you're relying on it to use stale cache, so this will break the test)